File: 387-FirstUniqueCharacterInAString/387-FirstUniqueCharacterInAString.js
// Last updated: 3/6/2026, 2:09:18 PM
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    for (let i = 0; i < s.length; i++) {
        if (s.indexOf(s[i]) === s.lastIndexOf(s[i])) {
            return i;
        }
    }
    return -1;
};

File: 387-FirstUniqueCharacterInAString/387-FirstUniqueCharacterInAString.cpp
// Last updated: 7/17/2026, 4:27:22 PM
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char,int> mp;

        for(char value : s){
            mp[value]++;
        }    

        for(int i=0;i<s.size();i++){
            if(mp[s[i]]==1){
                return i;
            }
        }

        return -1;
    }
};